Fix: always update cache components in editor

daleeidd opened the pull request:
Fix: #617 (partially)

Fixes an issue where the properties (max terrain height and resolution) on the depth cache or its transform (scale) are changed, the components (camera and textures) are not updated when in the editor when they should be. Users would have to enter and exit play mode which would cause confusion and possibly confusing bug reports. The update method is public so users can call during runtime if they need to change properties on the depth cache.

huwb reviewed (approved):
Thanks, good shout.

I wondered if the new code could/should be used during initial setup of the depth cache so that there is only one set of code to set the transform, camera params, etc. However not sure if this would work and I'm fine with merging this.

daleeidd commented:
Sounds good. Done. I have added an updateComponents param to PopulateCache. If true, then components are updated in InitObjects.

I have also added validation to check if the components are outdated and recommend user populates the cache.

The only thing I haven't handled is the scale. If we use the island scene as an example, and scale the parent object of both the cache and island, if the cache is baked, the cache will scale correctly. But if the cache is realtime, then it will distort. It is possible to correct this, but is tricky when the cache itself is rotated.

huwb commented:
Could we catch the scale issue with validation?

daleeidd commented:
Sure. I am now showing a validation error if the lossy scale is not uniform for X and Z when set to realtime. Rotating the cache seems fine though.

Also added switching the quad texture when the cache type is changed.

huwb commented:
Awesome.

Probably want to use Mathf.ApproximatelyEqual when checking the scale as doing exact comparison would be liable to fail.

daleeidd commented:
Done.

huwb commented:
Looks good

daleeidd commented:
Thanks. Merged.
